Vic2ToHoI4/Source/HOI4World/HoI4FocusTree.cpp

```diff
@@ -1504,70 +1503,57 @@ void HoI4FocusTree::addCommunistWarBranch(shared_ptr<HoI4::Country> Home,
 					warTargetCountryName.clear();
 				}
 
-				int v1 = rand() % 12 + 1;
-				int v2 = rand() % 12 + 1;
 				if (const auto& originalFocus = loadedFocuses.find("War"); originalFocus != loadedFocuses.end())
 				{
 					auto newFocus =
-						 originalFocus->second.makeTargetedCopy(Home->getTag(), warTargets[i]->getTag(), hoi4Localisations);
-					newFocus->id = "War" + warTargets[i]->getTag() + Home->getTag();
-					newFocus->available = "= {\n";
+						 originalFocus->second.makeTargetedCopy(Home->getTag(), warTarget->getTag(), hoi4Localisations);
+					newFocus->id = "War" + warTarget->getTag() + Home->getTag();
+					int v1 = rand() % 12 + 1;
+					int v2 = rand() % 12 + 1;
```

Comment:
While we're here, is it possible to replace these with something deterministic?

If not, at the least they should use C++-style random number generation: https://en.cppreference.com/w/cpp/numeric/random
There's examples of that elsewhere in the codebase.

Reply:
Sure, I'll make it dependant on relations value, like some other focuses. Although, even as it is, the date doesn't change between two conversions, not sure how that's accomplished.

Reply:
Unseeded random is rather deterministic, just in a mysterious manner. That's why we use is as a technique for selecting names and portraits out of a set. But no reason to be opaque here.